cpp/oneapi/dal/backend/memory.hpp

```diff
@@ -80,6 +175,71 @@ class usm_deleter {
     sycl::queue queue_;
 };
 
+template <typename T>
+using unique_usm_ptr = std::unique_ptr<T, usm_deleter<T>>;
+
+inline unique_usm_ptr<void> make_unique_usm(const sycl::queue& q,
+                                            std::size_t size,
+                                            sycl::usm::alloc alloc) {
+    return unique_usm_ptr<void>{ malloc(q, size, alloc), usm_deleter<void>{ q } };
+}
+
+inline unique_usm_ptr<void> make_unique_usm_device(const sycl::queue& q, std::size_t size) {
+    return unique_usm_ptr<void>{ malloc_device(q, size), usm_deleter<void>{ q } };
+}
+
+inline unique_usm_ptr<void> make_unique_usm_shared(const sycl::queue& q, std::size_t size) {
+    return unique_usm_ptr<void>{ malloc_shared(q, size), usm_deleter<void>{ q } };
+}
+
+inline unique_usm_ptr<void> make_unique_usm_host(const sycl::queue& q, std::size_t size) {
+    return unique_usm_ptr<void>{ malloc_host(q, size), usm_deleter<void>{ q } };
+}
+
+template <typename T>
+inline unique_usm_ptr<T> make_unique_usm(const sycl::queue& q,
+                                         std::int64_t count,
+                                         sycl::usm::alloc alloc) {
+    return unique_usm_ptr<T>{ malloc<T>(q, count, alloc), usm_deleter<T>{ q } };
+}
+
+template <typename T>
+inline unique_usm_ptr<T> make_unique_usm_device(const sycl::queue& q, std::int64_t count) {
+    return unique_usm_ptr<T>{ malloc_device<T>(q, count), usm_deleter<T>{ q } };
+}
+
+template <typename T>
+inline unique_usm_ptr<T> make_unique_usm_shared(const sycl::queue& q, std::int64_t count) {
+    return unique_usm_ptr<T>{ malloc_shared<T>(q, count), usm_deleter<T>{ q } };
+}
+
+template <typename T>
+inline unique_usm_ptr<T> make_unique_usm_host(const sycl::queue& q, std::int64_t count) {
+    return unique_usm_ptr<T>{ malloc_host<T>(q, count), usm_deleter<T>{ q } };
+}
+
+template <typename T>
+inline bool is_device_usm(const array<T>& ary) {
```

Comment:
it's related to array - probably better to move it closer to it

Reply:
I'm thinking to introduce such helpers to array at the `detail` level, but it should be planned and done in other PR.